services-js/registry-certs/client/store/OrderProvider.ts

```diff
@@ -30,28 +52,67 @@ export default class OrderProvider {
 
     const order = new Order(orderInfo, !!localStorage);
 
-    autorun(() => {
-      const { localStorage } = this;
-
-      if (!localStorage) {
-        return;
+    autorun(
+      () => {
+        const { localStorage } = this;
+
+        if (!localStorage) {
+          return;
+        }
+
+        if (order.info.storeContactAndShipping || order.info.storeBilling) {
+          const value = JSON.stringify(this.permanentStorageInfo(order.info));
+          localStorage.setItem(LOCAL_STORAGE_KEY, value);
+        } else {
+          localStorage.removeItem(LOCAL_STORAGE_KEY);
+        }
+      },
+      {
+        name: 'Order -> localStorage',
       }
-
-      if (order.info.storeContactAndShipping || order.info.storeBilling) {
-        const value = JSON.stringify(this.storableInfo(order.info));
-        localStorage.setItem(LOCAL_STORAGE_KEY, value);
-      } else {
-        localStorage.removeItem(LOCAL_STORAGE_KEY);
+    );
+
+    autorun(
+      () => {
+        const { sessionStorage } = this;
+
+        if (!sessionStorage) {
+          return;
+        }
+
+        // Unlike localStorage, the sessionStorage values are unfiltered and
+        // always saved.
+        const value = JSON.stringify(order.info);
+        sessionStorage.setItem(SESSION_STORAGE_KEY, value);
+      },
+      {
+        name: 'Order -> sessionStorage',
       }
-    });
+    );
 
     return order;
   }
 
-  // Returns a filtered OrderInfo based on the values of storeContactAndShipping
-  // and storeBilling. If they're true, includes the related values, '' if
-  // false.
-  storableInfo(info: OrderInfo): OrderInfo {
+  /**
+   * Removes the order from session storage. Call this after the order is
+   * submitted successfully.
+   */
+  clear() {
+    const { sessionStorage } = this;
+
+    if (sessionStorage) {
+      sessionStorage.removeItem(SESSION_STORAGE_KEY);
+    }
+  }
+
+  /**
+   * Returns a filtered OrderInfo based on the values of storeContactAndShipping
+   * and storeBilling. If they're true, includes the related values, '' if
```

Comment:
This comment is a little unclear; could we go with something along the lines of:

```suggestion
   * and storeBilling. If either is true, include its value, otherwise include an empty string
```